internal/http/config.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package http
+
+// TODO: Pull in other utilities from
+// https://github.com/signalfx/signalfx-agent/blob/master/pkg/core/common/httpclient/http.go.
+// HTTPConfig holds config options related to HTTP(S)
+type HTTPConfig struct {
```

Comment:
Happy to consider promoting this to the code repo, since now we also have OTLP/HTTP. @tigrannajaryan any opinion on this?

Reply:
Yes. I think it is best to let incubate for a while in Contrib and then it can be moved to core.